app/helpers/GeneralStatsHelper.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+namespace App\Helpers;
+use DateTime;
+use DateInterval;
+use Nette;
+use App\Model\Repository\Users;
+use App\Model\Repository\Groups;
+use App\Model\Repository\Exercises;
+use App\Model\Repository\Assignments;
+use App\Model\Repository\Solutions;
+use App\Model\Repository\AssignmentSolutionSubmissions;
+use App\Model\Repository\ReferenceSolutionSubmissions;
+use App\Model\Repository\SubmissionFailures;
+
+/**
+ * A structure with recorded global stats.
+ */
+class GeneralStats {
+  use Nette\SmartObject;
+
+  /**
+   * @var int|null
+   * Number of user created/registered during the selected period.
+   */
+  public $createdUsers = null;
+
+  /**
+   * @var int|null
+   * Total number of users at present time.
+   */
+  public $totalUsers = null;
+
+  /**
+   * @var int|null
+   * Number of users who could be deleted as inactive.
+   */
+  public $inactiveUsers = null;
+
+  /**
+   * @var int|null
+   * Total number of groups at present time (including archived).
+   */
+  public $totalGroups = null;
+
+  /**
+   * @var int|null
+   * Total number of archived groups at present time.
+   */
+  public $archivedGroups = null;
+
+  /**
+   * @var int|null
+   * Number of exercises created during the selected period.
+   */
+  public $createdExercises = null;
+  
+  /**
+   * @var int|null
+   * Total number of exercises at present time.
+   */
+  public $totalExercises = null;
+  
+  /**
+   * @var int|null
+   * Number of assignments created during the selected period.
+   */
+  public $createdAssignments = null;
+
+  /**
+   * @var int|null
+   * Total number of assignments at present time.
+   */
+  public $totalAssignments = null;
+
+  /**
+   * @var int|null
+   * Number of student solutions created during the selected period.
+   */
+  public $createdSolutions = null;
+
+  /**
+   * @var int|null
+   * Total number of student solutions at present time.
+   */
+  public $totalSolutions = null;
+
+  /**
+   * @var int|null
+   * Number of submissions (and resubmissions) during the selected period.
+   */
+  public $createdSubmissions = null;
+
+  /**
+   * @var int|null
+   * Number of failed submissions during the selected period.
+   */
+  public $failedSubmissions = null;
+}
+
+
+/**
+ * This helper gathers multiple statistics (typicaly numbers of new entites)
+ * over given period of time.
+ */
+class GeneralStatsHelper {
+  use Nette\SmartObject;
+
+  /** @var string */
+  private $inactivityThreshold;
+
+  /**
+   * @var Users
+   * @inject
```

Comment:
Inject? Really?

Reply:
Just copypasting your code :)

Reply:
Where from? In case this was in some helper, I shall be called idiot then...

Reply:
OK, maybe its not your code. I shall try to find and fix it at source as well.